### PR TITLE
Use storage.v1 instead of v1beta1

### PR DESCRIPTION
**Pull request by jsafrane**
storage.v1beta1 was used to work around GKE which did not expose v1. Now that GKE is updated, we can switch everything to v1.

This is simple sed v1beta1 -> v1 + enabled a new test + changed preference of exposed interfaces in `storage/install/install.go`.

@msau42, PTAL and let me know when GKE is updated with storage v1 API and this PR can be actually merged.

@kubernetes/sig-storage-pr-reviews 

```release-note
NONE
```

### CODE REVIEW

**Comment by jsafrane** [comment]
BTW, does it require an exception?

**Comment by msau42** [comment]
This change LGTM.  I just made a GKE-side change that hopefully fixes the failing GKE tests (issue was not due to storageclass changes).  It'll probably take a few hours for the change to rollout to the test infra, but if you could rerun the GKE tests then, it should give us good confidence in this change.

**Comment by msau42** [comment]
@k8s-bot cvm gke e2e test this

**Comment by msau42** [comment]
@k8s-bot gci gke e2e test this

**Comment by msau42** [comment]
@k8s-bot gci gke e2e test this

**Comment by msau42** [comment]
@k8s-bot cvm gke e2e test this

**Comment by jsafrane** [comment]
@msau42, I checked today with `KUBERNETES_PROVIDER=gke go run hack/e2e.go -v  --up` and GKE installed this way not expose storage.k8s.io/v1 API, thus tests in this PR fail.

```
$ kubectl version
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"2017-02-15T06:34:56Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}

**Comment by msau42** [comment]
Yes, because it's on version 1.5.3, while the default storage class is only allowed in 1.6+.  I'm pretty sure the Jenkins GKE tests exercise dynamic provisioning because earlier they were failing during the provisioning and now they are passing.  I'll try to find a dev cluster today so I can bring up 1.6 with your PR and run the storage tests.

File: test/e2e/volume_provisioning.go [hunk 1]
```diff
@@ -103,48 +103,46 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
...
+
 	framework.KubeDescribe("DynamicProvisioner Beta", func() {
 		It("should create and delete persistent volumes [Slow] [Volume]", func() {
 			framework.SkipUnlessProviderIs("openstack", "gce", "aws", "gke")
 
 			By("creating a StorageClass")
 			class := newStorageClass("", "beta")
-			_, err := c.StorageV1beta1().StorageClasses().Create(class)
-			defer c.StorageV1beta1().StorageClasses().Delete(class.Name, nil)
+			_, err := c.StorageV1().StorageClasses().Create(class)
```
Comment (msau42) [inline test/e2e/volume_provisioning.go]: Should the beta tests still use beta APIs?

**Comment by msau42** [comment]
I brought up a dev GKE cluster with your PR and ran all the dynamic provisioning tests + my new default sc ones (7 total) using the v1 API, and they all passed.

**Comment by jsafrane** [comment]
@msau42, thanks for the tests. How are we going to proceed with this PR? I'd like to have it merged in 1.6, so the v1 API is actually used and tested there. AFAIK, GKE tests are disabled for now (or at least are non-blocking), so we can merge this PR, right?

File: test/e2e/volume_provisioning.go (hunk 1 above)
Comment (jsafrane) [inline test/e2e/volume_provisioning.go]: Good idea, I added a new commit with that change, a bit more lines of code were necessary to use v1beta1 API.

**Comment by jsafrane** [comment]
@k8s-bot unit test this
flake: #39471

@k8s-bot kops aws e2e test this
@k8s-bot gci gce e2e test this
@k8s-bot cvm gce e2e test this

**Comment by msau42** [comment]
I agree we should try to get this into 1.6.  You're right that GKE tests are not queue-blocking anymore.  Regardless, I ran your PR on my dev GKE environment, so I am confident that they will not break this time :-)  And if they do, it'll be my job to fix them, not yours.

I pinged @ethernetdan and @idvoretskyi to see what they think.

**Comment by kerneltime** [comment]
cc @divyenpatel there are changes related to vSphere tests, can you take a look.

**Comment by divyenpatel** [comment]
We use e2e tests from master branch to validate 1.5 branch. so after this change, tests may fail to execute on new 1.5 based release?
cc: @BaluDontu

**Comment by msau42** [comment]
Yes, but how is that not a problem anytime you add a feature in a new release?

**Comment by BaluDontu** [comment]
@msau42 : Its not just for the 1.5 releases but for all releases where v1beta1 is being used. For eg: 1.5 and 1.4 release branches. 

Since the storage class version will now change to v1 from v1beta1, the same tests which would now get executed on master branch would fail on 1.5, 1.4 release branch as v1 is not supported there? Is there any solution for this?

**Comment by msau42** [comment]
I feel like the best approach would be to run 1.5 version tests against 1.5, 1.4 against 1.4, etc.  This seems like a generic testing problem that is not specific to storage classes.  How is any new feature supposed to write tests for if we need to add versioning checks everywhere?

**Comment by msau42** [comment]
I spoke to @ethernetdan and he suggested that the PR be reduced to only just test changes.

@divyenpatel also explained the Vsphere testing environment.  They have volume tests checked into 1.6 but not previous branches, so they are running the 1.6 version of the tests against all releases.  I've asked them to backport their tests to each release instead.  For this PR, we can also remove the vsphere test changes for now until that happens, so the vsphere tests will still use the beta APIs.  Still, if the vsphere tests end up using some common wrapper/library that we changed, then the vsphere tests may still encounter issues.

**Comment by erinboyd** [comment]
We should hold off on this until we have the ability to rev the tests as well to avoid having to revert this.

**Comment by msau42** [comment]
@erinboyd Do you see an issue with the main test framework, or just the vsphere ones?

**Comment by erinboyd** [comment]
Discussed with with @msau42 and with vsphere approval and Jan's fix, it should be ready to go. @jeffvance @copejon @saad-ali

**Comment by jsafrane** [comment]
Rebased, all e2e test changes were merged as #43296, so this PR is significantly smaller.

File: pkg/apis/storage/install/install.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func Install(groupFactoryRegistry announced.APIGroupFactoryRegistry, registry *r
 		&announced.GroupMetaFactoryArgs{
 			GroupName: storage.GroupName,
 			// TODO:  change the order when GKE supports v1
```
Comment (msau42) [inline pkg/apis/storage/install/install.go]: This comment can be removed

File: test/e2e/volume_provisioning.go [hunk 3]
```diff
@@ -107,7 +107,7 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
 
 	framework.KubeDescribe("DynamicProvisioner", func() {
 		It("should create and delete persistent volumes [Slow] [Volume]", func() {
-			framework.SkipUnlessProviderIs("openstack", "gce", "aws", "gke")
```
Comment (msau42) [inline test/e2e/volume_provisioning.go]: @divyenpatel is the vsphere CI ready, or should the vsphere changes be left out?

**Comment by msau42** [comment]
@k8s-bot gce etcd3 e2e test this

File: test/e2e/volume_provisioning.go (hunk 3 above)
Comment (divyenpatel) [inline test/e2e/volume_provisioning.go]: @msau42 Lets add vsphere to the supported provider list for this test.
For Internal vSphere CI, we can use local branch copy, until I cross port all our tests to respective active branches.

cc: @tusharnt, @BaluDontu

File: pkg/apis/storage/install/install.go (hunk 2 above)
Comment (jsafrane) [inline pkg/apis/storage/install/install.go]: removed

**Comment by jsafrane** [comment]
@k8s-bot bazel test this

**Comment by jsafrane** [comment]
@thockin, PTAL and approve

**Comment by jsafrane** [comment]
rebased

**Comment by jsafrane** [comment]
@thockin, please approve

**Comment by thockin** [comment]
@k8s-bot test this

**Comment by msau42** [comment]
W0424 13:55:30.870] # k8s.io/kubernetes/test/e2e/storage
W0424 13:55:30.871] test/e2e/storage/vsphere_volume_fstype.go:97: cannot use storageClassSpec (type *"k8s.io/kubernetes/pkg/apis/storage/v1".StorageClass) as type *"k8s.io/kubernetes/pkg/apis/storage/v1beta1".StorageClass in argument to client.StorageV1beta1().StorageClasses().Create
W0424 13:55:30.871] test/e2e/storage/vsphere_volume_fstype.go:101: cannot use storageclass (type *"k8s.io/kubernetes/pkg/apis/storage/v1beta1".StorageClass) as type *"k8s.io/kubernetes/pkg/apis/storage/v1".StorageClass in argument to getVSphereClaimSpecWithStorageClassAnnotation

**Comment by jsafrane** [comment]
vsphere fixed

**Comment by jsafrane** [comment]
@k8s-bot unit test this

**Comment by jsafrane** [comment]
@k8s-bot cvm gce e2e test this
@k8s-bot gci gce e2e test this
@k8s-bot non-cri e2e test this

**Comment by jsafrane** [comment]
integration tests fixed (I hope)

**Pull request merged**